doc/manual/manual.md

```diff
@@ -2648,6 +2650,88 @@ construct compound values:
 * `mir_tuple_value : [MIRValue] -> MIRValue` construct a tuple with the given
   list of values as elements.
 
+### MIR slices
+
+Slices are a unique form of compound type that is currently only used during
+MIR verification. Unlike other forms of compound values, such as arrays, it is
+not possible to directly construct a slice. Instead, one must take a slice of
+an existing reference value that points to the thing being sliced. The
+following commands are used to construct slices:
+
+* `mir_slice_value : MIRValue -> MIRValue`: the SAWScript expression
+  `mir_slice_value base` is equivalent to the Rust expression `base[..]`,
+  i.e., a slice of the entirety of `base`. `base` must be a reference to
+  an array value.
+* `mir_slice_range_value : MIRValue -> Int -> Int -> MIRValue`: the SAWScript
+  expression `mir_slice_range_value base start end` is equivalent to the Rust
+  expression `base[start..end]`, i.e., a slice over a part of `base` which
+  ranges from `start` to `end`. `base` must be a reference to an array value,
+  and `start` and `end` are assumed to be zero-indexed. `start` must not exceed
+  `end`, and `end` must be strictly less than the length of the array that
+  `base` points to.
```

Comment:
In Rust, `start..end` is a half-open range, so `end == len` is valid.  For example:
```Rust
let arr = [1, 2, 3, 4, 5];
let slice = &arr[0..5];
```
`slice` is a 5-element slice containing all the elements of `arr`.

Reply:
Another occurrence of my earlier typo leaking out. I've corrected this to say "`end` must not exceed the length of the array".